L1Trigger/L1THGCalUtilities/python/hgcalTriggerChains.py

```diff
@@ -90,14 +118,80 @@ def create_sequences(self, process):
             if ntuple!='' and not hasattr(process, ntuple_name):
                 setattr(process, ntuple_name, self.ntuple[ntuple](process, ntuple_inputs))
                 ntuple_sequence *= getattr(process, ntuple_name)
+        for vfe,calo,selector,ntuple in self.truth_chain:
+            calo_name = '{0}{1}'.format(vfe, calo)
+            ntuple_name = '{0}{1}{2}'.format(vfe, calo, ntuple)
+            if selector!='' and not hasattr(process, selector_name):
+                setattr(process, selector_name, self.selector[selector](process, calo_name))
+                selector_sequence *= getattr(process, selector_name)
+            if selector=='':
+                ntuple_inputs = [calo_name,
+                            '',
```

Comment:
For empty strings, there should be some checks added somewhere in case the user request the `clusters` ntuple in the ntuple list

Reply:
I'm not sure about this part. I changed the empty strings to "calo_name" for now. This chain doesn't run any 2d or 3d clustering, so I don't know what it would mean for the user to request those collections.

Reply:
Actually I don't think using simply `truth_prod_name` would work, because `:HGCalBackendLayer1Processor2DClustering` and `:HGCalBackendLayer2Processor3DClustering` are needed in the input tag.
But what I was saying is that if empty strings are used here checks should be added for instance in `create_ntuple' in case the inputs and ntuple list are not compatible.